exporters/etw/include/opentelemetry/exporters/etw/etw_properties.h

```diff
@@ -0,0 +1,476 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include "opentelemetry/version.h"
+
+#include "opentelemetry/common/key_value_iterable_view.h"
+
+#include <opentelemetry/nostd/span.h>
+#include <map>
+#include <string>
+#include <vector>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace core
+{};
+namespace trace
+{};
+namespace exporter
+{
+namespace ETW
+{
+
+/**
+ * @brief PropertyVariant provides:
+ * - a constructor to initialize from initializer lists
+ * - an owning wrapper around `common::AttributeValue`
+ */
+using PropertyVariant =
+    nostd::variant<bool,
+                   int32_t,
+                   int64_t,
+                   uint32_t,
+                   uint64_t,
+                   double,
+                   std::string,
+#ifdef HAVE_CSTRING_TYPE
+                   const char *,
+#endif
+#ifdef HAVE_SPAN_BYTE
+                   // TODO: 8-bit byte arrays / binary blobs are not part of OT spec yet!
+                   // Ref: https://github.com/open-telemetry/opentelemetry-specification/issues/780
+                   std::vector<uint8_t>,
+#endif
+                   std::vector<bool>,
+                   std::vector<int32_t>,
+                   std::vector<int64_t>,
+                   std::vector<uint32_t>,
+                   std::vector<uint64_t>,
+                   std::vector<double>,
+                   std::vector<std::string>>;
+
+/**
+ * @brief PropertyValue class that holds PropertyVariant and
+ * provides converter for non-owning common::AttributeValue
+ */
+class PropertyValue : public PropertyVariant
+{
+
+  /**
+   * @brief Convert span<T> to vector<T>
+   * @tparam T
+   * @param source
+   * @return
+   */
+  template <typename T>
+  static std::vector<T> to_vector(const nostd::span<const T, std::size_t(-1)> &source)
+  {
+    return std::vector<T>(source.begin(), source.end());
+  };
+
+  /**
+   * @brief Convert span<string_view> to vector<string>
+   * @param source
+   * @return
+   */
+  std::vector<std::string> static to_vector(const nostd::span<const nostd::string_view> &source)
+  {
+    std::vector<std::string> result;
+    result.reserve(source.size());
+    for (const auto &item : source)
+    {
+      result.push_back(std::string(item.data()));
+    }
+    return result;
+  };
+
+  /**
+   * @brief Convert vector<INTEGRAL> to span<INTEGRAL>.
+   * @tparam T
+   * @param std::vector<T>
+   * @return nostd::span<const T>
+   */
+  template <typename T, std::enable_if_t<std::is_integral<T>::value, bool> = true>
+  static nostd::span<const T> to_span(const std::vector<T> &vec)
+  {
+    nostd::span<const T> result(vec.data(), vec.size());
+    return result;
+  };
+
+  /**
+   * @brief Convert vector<FLOAT> to span<const FLOAT>.
+   * @tparam T
+   * @param std::vector<T>
```

Comment:
Should the comments like `@param` be fixed with the actual parameter name?

Reply:
Yes